guacamole/src/main/webapp/app/navigation/directives/guacUserMenu.js

```diff
@@ -134,6 +134,45 @@ angular.module('navigation').directive('guacUserMenu', [function guacUserMenu()
                 return authenticationService.isAnonymous();
             };
 
+            /**
+             * Requests fullscreen for the Guacamole page.
+             */
+            $scope.fullscreen = function fullscreen() {
+                var elem = document.documentElement;
+
+                if (!elem.fullscreenElement && !elem.mozFullScreenElement &&
+                    !elem.webkitFullscreenElement && !elem.msFullscreenElement) {
+                    if (elem.requestFullscreen) {
+                        elem.requestFullscreen();
+                    } else if (elem.mozRequestFullScreen) { /* Firefox */
+                        elem.mozRequestFullScreen();
+                    } else if (elem.webkitRequestFullscreen) { /* Chrome, Safari and Opera */
+                        elem.webkitRequestFullscreen();
+                    } else if (elem.msRequestFullscreen) { /* IE/Edge */
+                        elem.msRequestFullscreen();
+                    }
+                } else {
+                    if (elem.exitFullscreen) {
+                        elem.exitFullscreen();
+                    } else if (elem.mozCancelFullScreen) { /* Firefox */
+                        elem.mozCancelFullScreen();
+                    } else if (elem.webkitExitFullscreen) { /* Chrome, Safari and Opera */
+                        elem.webkitExitFullscreen();
+                    } else if (elem.msExitFullscreen) { /* IE/Edge */
+                        elem.msExitFullscreen();
+                    }
+                }
```

Comment:
There may be a cleaner way to shim this. For example:

https://github.com/apache/guacamole-client/blob/b51d2a88b8415738f6fff19cc99bc2b3e4690da2/guacamole-common-js/src/main/webapp/modules/AudioRecorder.js#L189-L197

Reply:
Interesting! I will test if I can shim this like you proposed.Thanks.

Reply:
@mike-jumper That should be good now